inngest-core/src/main/kotlin/com/inngest/Environment.kt

```diff
@@ -0,0 +1,13 @@
+package com.inngest
+
+object Environment {
+    fun inngestHeaders(framework: SupportedFrameworkName? = null): RequestHeaders {
+        val sdk = "inngest-kt:${Version.getVersion()}"
+        return mapOf(
+            InngestHeaderKey.ContentType.value to "application/json",
+            InngestHeaderKey.Sdk.value to sdk,
+            InngestHeaderKey.UserAgent.value to sdk,
+            InngestHeaderKey.Framework.value to (framework?.value),
+        ).filterValues { (it is String) }.entries.associate { (k, v) -> k to v!! }
```

Comment:
is this to check for null values? Why is `SupportedFrameworkName` nullable anyway, could we make it not nullable?

Reply:
it's nullable because the Inngest's client uses it and isn't provided the framework option as mentioned ☝️ by Darwin: [inngest.com/docs/reference/client/create](https://www.inngest.com/docs/reference/client/create).

also looking at these two header generation functions in the other sdks, i think we'll have more optional headers implemented:

https://github.com/inngest/inngest-py/blob/2dd0ebd45c2fdbb035c03e0ec188567a54e76c31/inngest/_internal/client_lib.py#L137-L144
https://github.com/inngest/inngest-js/blob/1b9f101ca6cd310e429cebcf8bceaf2faf309624/packages/inngest/src/helpers/env.ts#L297-L301

Reply:
fair enough, I guess it's possible people will want to use the Inngest client directly without a framework adapter